src/common.jl

```diff
@@ -0,0 +1,161 @@
+abstract ODEIterAlgorithm <: AbstractODEAlgorithm
+immutable feuler <: ODEIterAlgorithm end
+immutable rk23 <: ODEIterAlgorithm end
+immutable rk45 <: ODEIterAlgorithm end
+immutable feh78 <: ODEIterAlgorithm end
+immutable ModifiedRosenbrock <: ODEIterAlgorithm end
+immutable midpoint <: ODEIterAlgorithm end
+immutable heun <: ODEIterAlgorithm end
+immutable rk4 <: ODEIterAlgorithm end
+immutable feh45 <: ODEIterAlgorithm end
+
+typealias KW Dict{Symbol,Any}
+
+function solve{uType,tType,isinplace,algType<:ODEIterAlgorithm,F}(prob::AbstractODEProblem{uType,tType,isinplace,F},
+    alg::algType,timeseries=[],ts=[],ks=[];dense=true,save_timeseries=true,
+    saveat=[],callback=()->nothing,timeseries_errors=true,dense_errors=false,
+    kwargs...)
+  tspan = prob.tspan
+
+  if tspan[end]-tspan[1]<tType(0)
```

Comment:
>What about the sort(unique(...)) thing?

That's for "saveat" style chosen outputs. The things I tried didn't work, so I just don't have saveat implemented at all in this.